ui/lesson/HashChallenge/HashChallenge.tsx

```diff
@@ -0,0 +1,151 @@
+'use client'
+
+import { useEffect, useState } from 'react'
+import { Lesson, LessonTabs, LessonPrompt, StatusBar, Tooltip } from 'ui'
+import LessonHasher from './HashPrompt'
+import clsx from 'clsx'
+
+/**
+ * @answer {string} correct answer to the challenge problem
+ * @next {string} link to next part of chapter
+ * @label {string} label on top of input field
+ * @pattern {RegEx} pattern for input field
+ * @hints {boolean} determine whether the input field displays hints for the user
+ */
+export default function HashChallenge({
+  // children,
+  answer,
+  answerHint,
+  next,
+  label,
+  label2,
+  auto,
+  hints,
+}: {
+  // children: any
+  answer: string | number
+  next: any
+  label: string
+  label2: string
+  answerHint?: string
+  auto?: boolean
+  hints?: boolean
+}) {
+  const [input, setInput] = useState('')
+  const [userInput, setUserInput] = useState('')
+  const [hintTimer, setHintTimer] = useState(false)
+
+  const inputView = (
+    <>
+      hintTimer && (
+      <Tooltip
+        id={`challenge-hint-tooltip}`}
+        position="top"
+        offset={0}
+        content={
+          <div className="flex flex-col">
+            <span className="whitespace-nowrap">Need a hint?</span>
+          </div>
+        }
+      >
+        {input}
+      </Tooltip>
+      ) || input
+    </>
+  )
+
+  const handleChange = (event) => {
+    setInput(event.target.value)
+  }
+
+  useEffect(() => {
+    const hintTimeout = () => {
+      setTimeout(() => {
+        setHintTimer(true)
+      }, 15000)
+    }
+    answerHint && hintTimeout()
+  })
```

Comment:
You are not clearing the timeout in the `useEffect`'s return function, there is also no dependency array so this effect will trigger on every render, if that is by design please explain that with a comment.

Reply:
I intend for the hintTooltip to show after 15 seconds, with the timer starting only after the user begins to put in an input> I have made <s>some</s> many adjustments to the useEffect and I think I got it!